Frends.AzureDataLake.DeleteDirectory/Frends.AzureDataLake.DeleteDirectory.Tests/UnitTests.cs

```diff
@@ -0,0 +1,137 @@
+    {
+        await AzureDataLake.DeleteDirectory(new Input { ConnectionString = _connectionString, ContainerName = null, DirectoryName = _directoryName }, new Options { ThrowErrorIfDirectoryDoesNotExists = false }, new CancellationToken());
+        await AzureDataLake.DeleteDirectory(new Input { ConnectionString = null, ContainerName = _containerName, DirectoryName = _directoryName }, new Options { ThrowErrorIfDirectoryDoesNotExists = false }, new CancellationToken());
+        await AzureDataLake.DeleteDirectory(new Input { ConnectionString = _connectionString, ContainerName = _containerName, DirectoryName = null }, new Options { ThrowErrorIfDirectoryDoesNotExists = false }, new CancellationToken());
+    }
+
+    [TestMethod]
+    [ExpectedException(typeof(Exception))]
+    public async Task TestDeleteDirectory_throws_ParameterNotValid()
+    {
+        await AzureDataLake.DeleteDirectory(new Input { ConnectionString = "Not valid parameter", ContainerName = "Valid name", DirectoryName = "Valid name" }, new Options { ThrowErrorIfDirectoryDoesNotExists = false }, new CancellationToken());
+        await AzureDataLake.DeleteDirectory(new Input { ConnectionString = "name=value", ContainerName = "Valid name", DirectoryName = "Valid name" }, new Options { ThrowErrorIfDirectoryDoesNotExists = false }, new CancellationToken());
+    }
+
+    [TestMethod]
+    [ExpectedException(typeof(Exception))]
+    public async Task TestDeleteDirectory_throws_ClientNotFound()
+    {
+        await AzureDataLake.DeleteDirectory(new Input { ConnectionString = "DefaultEndpointsProtocol=https;AccountName=unitTestStorage;AccountKey=abcdefghijklmnopqrstuyxz123456789;EndpointSuffix=core.windows.net", ContainerName = _containerName, DirectoryName = _directoryName }, new Options { ThrowErrorIfDirectoryDoesNotExists = false }, new CancellationToken());
+    }
+
+    [TestMethod]
+    public async Task AccessTokenAuthenticationTest()
+    {
+        var containerName = $"test{Guid.NewGuid()}";
+        var directoryName = $"test{Guid.NewGuid()}";
+
+        input = new Input
+        {
+            ConnectionMethod = ConnectionMethod.OAuth2,
+            ContainerName = containerName,
+            DirectoryName = directoryName,
+            StorageAccountName = _storageAccount,
+            ApplicationID = _appID,
+            TenantID = _tenantID,
+            ClientSecret = _clientSecret
+        };
+
+        var options = new Options
+        {
+            ThrowErrorIfDirectoryDoesNotExists = false
+        };
+
+        DataLakeFileSystemClient container = new DataLakeServiceClient(_connectionString).GetFileSystemClient(input.ContainerName);
+        await container.CreateIfNotExistsAsync(null, new CancellationToken());
+
+        var directoryClient = AzureDataLake.GetDataLakeDirectory(input);
+        await directoryClient.CreateIfNotExistsAsync(null, new CancellationToken());
+
+        var result = await AzureDataLake.DeleteDirectory(input, options, default);
```

Comment:
lets use cancellationtoken.none instead of default. this is a bit more verbose and shows that the third parameter is a cancellation token. also, lets try to follow the same style across code. So e.g. if we use cancellationtoken.none, then use it in all places.